tests/test_decode.py

```diff
@@ -1,7 +1,22 @@
 """Test cases for the chardetng_py module."""
-from chardetng_py import decode
+import pytest
+from chardetng_py import detect
 
 
 def test_decode() -> None:
     """Very basic decode litmus test."""
-    assert decode(b"Jakby r\xeaka Boga") == "Jakby rêka Boga"
+    raw = b"Jakby r\xeaka Boga"
+    assert raw.decode(detect(raw)) == "Jakby rêka Boga"
+
+
+@pytest.mark.parametrize(("text", "encoding", "detected_encoding"), [
+    ("สวัสดี, โลก", "cp874", "cp874"),
+    ("员会 ⛆", "gb18030", "gb18030"),
+])
+def test_returns_python_supported_names(
+    text: str,
+    encoding: str,
+    detected_encoding: str,
+) -> None:
+    """Ensure detection returns encoding names that are supported in Python."""
+    assert detect(text.encode(encoding)) == detected_encoding
```

Comment:
Quite nice. This is actual logic that we have bolted on top of the rust library, so it makes sense to test it.